### DISCUSSION TITLE
Does Copy-PnPFile copy all settings?

### DISCUSSION THREAD

**Discussion by gitulm** [pnp/powershell#4636] [Q&A]
We have about 100 sites that should have the same document libraries. I copied the sample library with

`Copy-PnPList -Identity $LibraryName -DestinationWebUrl $einr -Title $LibraryName`

All columns were copied, everything was correct. Then I copied the sample documents with

`Copy-PnPFile -SourceUrl $_.ServerRelativeUrl -TargetUrl $TargetLibraryServerRelativeUrl -Force -OverwriteIfAlreadyExists`

All documents were copied correctly and they also had the correct column settings.

In all sites there should be a sample page where a source sample page is copied and in this sample page there are several "Highlighted Content" web parts that refer to the copied library and its columns. I copied the sample page with

`Copy-PnPFile -SourceUrl "/sites/$tempquelle/sitepages/Interne-Dokumente.aspx" -TargetUrl "https://xxx.sharepoint.com$einr2/sitepages" -force -Overwrite -IgnoreVersionHistory`

The page is copied, but the display in the "Highlighted Content" area is not adopted. Are such settings not made during the copying process or is there an error?``

**Comment by waaromikniet** [comment]
I tested this and for me it works. I did specify the page name in my targetUrl:

`Copy-PnPFile -SourceUrl "/sites/Ontwikkel/SitePages/SourcePage.aspx" -TargetUrl "/sites/Ontwikkel/SitePages/TargetPage.aspx" -Overwrite -Force -IgnoreVersionHistory`

I copied on the same site. When you copy to another site maybe the webpart references the library based on ID and its not the same on the target site?

What version of PnP.PowerShell are you using?

**Comment by gitulm** [comment]
I use PnP.Powershell version 2.12.0
As I have just noticed, when the page is copied, the first document library of alle libraries is used for the "Highlighted Content" web part. So if the first library starts with A, then it takes that. On the copied page, I set on the "Highlighted Content" webpart to a library that starts with Z. That's why this was not adopted during the copying process. Is this a bug or a feature?

**Reply by waaromikniet** [reply]
I would say it's a missing feature to keep the correct document library name. 
Have you tried exporting the page to a template and see if the library name is extracted? 
Maybe you can use a template to provision the page.

**Comment by gitulm** [comment]
The template is very large and unreadable for me, so I will make the settings manually for the copied pages.
